### PR TITLE
Video thumbnail

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Enable video thumbnail.
Needed:
'enabledPreviewProviders' => array(
'OC\Preview\Movie'
)
in config.php

@jancborchardt we need a video thumbnail.

### CODE REVIEW

**Comment by jancborchardt** [comment]
Nice! :) But does this mean that every single person needs to add this config option to their config? Can’t we enable that by default? cc @LukasReschke

**Comment by Spacefish** [comment]
@jancborchardt you need to activate this in your server, but this patch is needed to even be able to see videothumbs in the android app +1
Enabling the Video Preview Provider by default would be nice, but we should keep security in mind. There are multiple bugs lately in ffmpeg which allowed code execution due to malformed container / bytestream headers.. Just think of the stagefright bug in android.

File: src/com/owncloud/android/ui/fragment/OCFileListFragment.java [hunk 1]
```diff
@@ -720,6 +720,7 @@ private void updateLayout() {
                 registerLongClickListener();
             } else {
                 switchToListView();
+//                switchToGridView();
```
Comment (przybylski) [inline src/com/owncloud/android/ui/fragment/OCFileListFragment.java]: if unused, please remove

**Comment by tobiasKaminsky** [comment]
@jancborchardt not every user, but every admin.
I have talked to @LukasReschke on IRC an he says that this would be a security problem as the previews are currently not generated in a sandbox.

File: src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java [hunk 2]
```diff
@@ -389,6 +404,51 @@ public static ThumbnailGenerationTask getBitmapWorkerTask(ImageView imageView) {
...
+
+        // compute visual center of play button, according to resized image
+        int x1 = resizedPlayButton.getWidth();
+        int y1 = resizedPlayButton.getHeight() / 2;
+        int x2 = 0;
+        int y2 = resizedPlayButton.getWidth();
+        int x3 = 0;
+        int y3 = 0;
+
+        double ym = ( ((Math.pow(x3,2) - Math.pow(x1,2) + Math.pow(y3,2) - Math.pow(y1,2)) *
```
Comment (przybylski) [inline src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: a lot of zeros in the calculations so this probably can be simplified ;)

**Comment by jancborchardt** [comment]
> not every user, but every admin.

@tobiasKaminsky yeah, this is still a bit bad cause this results in different UX for different people depending on the server, and this should ideally not happen. Since it’s a security issue, is it possible to generate the previews client-side as well?

File: src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java (hunk 2 above)
Comment (tobiasKaminsky) [inline src/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: I doubt so, as this is a mainly pythagoras and he needs three coordinates. 
To simplify it, I have chosen them to be mostly zero.

**Comment by tobiasKaminsky** [comment]
@jancborchardt when the previews are generated in a sandbox the generation of videos can be enabled by default.
The fact that it is a security risk we should not enable preview on client. Beside the security risk, it is also very difficult as the video has to be downloaded prior to generate a preview locally.
So I think we should do it this way and provide previews for those (home) servers that enable the preview generation and for the others we should provide a good video thumbnail.

**Comment by jancborchardt** [comment]
Okidoke, sounds good

**Comment by tobiasKaminsky** [comment]
So this can go in after we have a dummy video thumbnail @jancborchardt (the third reminder)

**Comment by AndyScherzinger** [comment]
How should the dummy look like? We do have an video icon which we could use as long as we do not have created a thumbnail of the video,  can we not?

**Comment by jancborchardt** [comment]
@tobiasKaminsky aaah I didn’t know that was a request! I assumed you already have the dummy video thumbnail in the filetypes, because it’s here: https://github.com/nextcloud/server/blob/master/core/img/filetypes/video.svg

**Comment by jancborchardt** [comment]
And for when a thumbnail is created, didn’t we already have the play icon overlayed on top of it for the grid view? cc @AndyScherzinger

**Comment by AndyScherzinger** [comment]
@jancborchardt as far as I recall we do, yes.

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger can you make video thumbnails for me?
@jancborchardt yes, play button as overlay on video thumbnails, both in list and grid view

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky absolutely, do you need an icon which is shown when there is no thumbnail or a "play icon" overlay which is put on top of a video thumbnail? (can do both, whatever you need :smile:)

**Comment by tobiasKaminsky** [comment]
Placeholder when there is no video based thumbnail available.

**Comment by AndyScherzinger** [comment]
Erm...  It is all in place :D
https://github.com/nextcloud/android/blob/master/res/drawable-hdpi/file_movie.png

**Comment by AndyScherzinger** [comment]
These are already in use for all videos atm since we didn't have video thumbnails yet.  :)

**Comment by tobiasKaminsky** [comment]
You are right, then we are done here

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky can you rebase and resolve the conflict? I can also give it a try but I guess it is saver when you do it since it is your PR. I'll test drive it afterwards. Code changes look good to me so far :)

**Comment by AndyScherzinger** [comment]
The placeholders work fine for me, the real video thumbnails need to be checked by someone else :(

device-2016-07-02-181725

**Comment by tobiasKaminsky** [comment]
This is already working :)
But we have to get this running on your pc ;-)

**Comment by AndyScherzinger** [comment]
Works, see screenshot. But only if the server already created the thumbnail. At first I didn't log into the web view and opened the folder which seems to trigger the thumbnail generation. After I did that the Android client was able to fetch the thumbnails.

device-2016-07-14-223419

**Comment by AndyScherzinger** [comment]
:shipit:

**Comment by AndyScherzinger** [comment]
@przybylski do you have the time to do a review? I already did one :)

**Comment by przybylski** [comment]
@AndyScherzinger I already did a review some time ago, if nothing changed since than 👍

**Comment by AndyScherzinger** [comment]
Sorry @przybylski I missed that :(

@tobiasKaminsky can you update it to latest master for another test round before merging?

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger is updated

**Comment by AndyScherzinger** [comment]
Sweet, will do a test tomorrow or beginning of next week.

**Comment by AndyScherzinger** [comment]
Test successful, can imho be merged. :+1:

**Comment by AndyScherzinger** [comment]
LGTM is broken again... grrr

**Comment by AndyScherzinger** [comment]
:+1:

**Pull request merged**